hbase-common/src/main/java/org/apache/hadoop/hbase/io/encoding/HFileBlockDefaultDecodingContext.java

```diff
@@ -103,6 +103,11 @@ public void prepareDecoding(int onDiskSizeWithoutHeader, int uncompressedSizeWit
             compression.createDecompressionStream(dataInputStream, decompressor, 0)) {
             BlockIOUtils.readFullyWithHeapBuffer(is, blockBufferWithoutHeader,
               uncompressedSizeWithoutHeader);
+            // The buffer has allocated space for checksum, but it's not always filled. That might
```

Comment:
Yeah, I see this is getting allocated in HFileBlock#allocateBuffer. The thing I'm left wondering if why `BlockIOUtils.readFullyWithHeapBuffer(..)` is only _sometimes_ copying the checksum bytes after the uncompressed datablock. I guess that is the outstanding question :)

Reply:
I agree we should understand that.